File: Java Presidents Project/app/src/main/java/com/example/javapresidents/AddEditOne.java
//****************************************************************************

package com.example.javapresidents;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import java.util.List;


public class AddEditOne extends AppCompatActivity
{

    Button btnOK;
    Button btnCancel;

    EditText etPresDate, etPresName, etPresImageURL;

    List<President> presidentList;
    TextView tvPresId;
    int id;
    MyJavaClass myApplication = (MyJavaClass) this.getApplication();

    //************************************************************************

    /**@methodName:
     * @param:
     * @return:
     * @description:
     * */

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_edit_one);

        presidentList = MyJavaClass.getPresidentList();

        btnOK = findViewById(R.id.btn_OK);
        btnCancel = findViewById(R.id.btn_Cancel);
        Intent mainActivityIntent =
                new Intent(AddEditOne.this, MainActivity.class);

        etPresName = findViewById(R.id.editTextTextPersonName);
        etPresDate = findViewById(R.id.editTextDate);
        etPresImageURL = findViewById(R.id.et_URL);
        tvPresId = findViewById(R.id.tvPresIDNumber);

        Intent intent = getIntent();
        id = intent.getIntExtra("id", -1);
        President president = null;

        if (id >= 0)
        {
            // Edit president
            for (President p : presidentList)
            {
                if (p.getId() == id)
                {
                    president = p;
                } // End if
            } // End for

            etPresName.setText(president.getName());
            etPresDate.setText(
                    String.valueOf(president.getDateAssumedOffice()));
            etPresImageURL.setText(president.getImageURL());
            tvPresId.setText(String.valueOf(id));
        } // End if


        btnOK.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View view)
            {
                if (id >= 0)
                {
                    // Update
                    President updatedPres =
                            new President(id, etPresName.getText().toString(),
                                          Integer.parseInt(etPresDate.getText()
                                                                     .toString()),
                                          etPresImageURL.getText().toString());
                    presidentList.set(id, updatedPres);
                } // End if

                else
                {
                    // Create president object
                    int nextID = MyJavaClass.getNextInt();
                    President newPresident = new President(nextID,
                                                           etPresName.getText()
                                                                     .toString(),
                                                           Integer.parseInt(
                                                                   etPresDate.getText()
                                                                             .toString()),
                                                           etPresImageURL.getText()
                                                                         .toString());

                    // Add president to list
                    presidentList.add(newPresident);
                    myApplication.setNextInt(nextID++);
                } // End else
                // Go back to MainActivity
                startActivity(mainActivityIntent);
            } // End onClick
        }); // btnOK.setOnClickListener

        btnCancel.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View view)
            {
                startActivity(mainActivityIntent);
            } // End onClick
        }); // End btnCancel.setOnClickListener
    } // End onCreate
} // End AddEditOne class

File: app/src/main/java/com/example/presidentlist/AddEditOne.kt
package com.example.presidentlist

import android.content.Intent
import android.os.Bundle
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import java.lang.String
import kotlin.Int


class AddEditOne : AppCompatActivity()
{
    
    var presidentList: MutableList<President>? = null
    var id = 0
    
    
    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_edit_one)
        
        presidentList = MyKotlinClass.getPresidentList()
        
        val btnOK: Button = findViewById(R.id.btn_OK)
        val btnCancel: Button = findViewById(R.id.btn_Cancel)
        val mainActivityIntent =
            Intent(this@AddEditOne, MainActivity::class.java)
        val etPresName: EditText = findViewById(R.id.editTextTextPersonName)
        val etPresDate: EditText = findViewById(R.id.editTextDate)
        val etPresImageURL: EditText = findViewById(R.id.et_URL)
        val tvPresId: TextView = findViewById(R.id.tvPresIDNumber)
        
        
        val intent = intent
        id = intent.getIntExtra("id", -1)
        var president: President? = null
        
        if (id >= 0)
        {
            // Edit president
            for (p in presidentList!!)
            {
                if (p.id === id)
                {
                    president = p
                } // End if
            } // End for
            
            if (president != null)
            {
                etPresName.setText(president.presName)
                etPresDate.setText(String.valueOf(president.dateAssumedOffice))
                etPresImageURL.setText(president.imageURL)
                tvPresId.text = id.toString()
            } // End if
        } // End if
        
        
        btnOK.setOnClickListener { v ->
            
            if (id >= 0)
            {
                // Update
                val updatedPres = President(id, etPresName.text.toString(),
                    etPresDate.text.toString().toInt(),
                    etPresImageURL.text.toString())
                presidentList!![id] = updatedPres
            } // End if
            
            else
            {
                // Create president object
                var nextID: Int = MyKotlinClass.nextInt
                val newPresident = President(nextID, etPresName.text.toString(),
                    etPresDate.text.toString().toInt(),
                    etPresImageURL.text.toString())
                
                // Add president to list
                presidentList!!.add(newPresident)
                MyKotlinClass.nextInt = nextID++
                //myApplication.setNextInt(nextID++)
            } // End else
            
            startActivity(mainActivityIntent)
        }
        
        btnCancel.setOnClickListener { _ ->
            startActivity(mainActivityIntent)
        }
    } // End onCreate
} // End AddEditOne class
